libs/data-access-recipes/src/lib/data-access-recipes/+state/recipes.effects.ts

```diff
@@ -28,5 +29,59 @@ const loadAll$ = createEffect(
   { functional: true },
 );
 
+const createRecipe$ = createEffect(
+  (actions$ = inject(Actions)) => {
+    const repo = inject(RecipesRepository);
+    const dialog = inject(MatDialog);
+
+    return actions$.pipe(
+      ofType(RecipesUiActions.createRecipe),
+      exhaustMap((action) => {
+        return repo.create(action.recipeDto).pipe(
+          map((res) => {
+            dialog.closeAll();
```

Comment:
it shouldn't be here; dialog is an implementation detail, so we'd rather not add any unrelated dependencies. Imagine needing to move your forms from a dialogue to regular components. That would necessitate adjustments to the effects logic, an unnecessary complication if we maintain a streamlined and independent architecture.

Reply:
@DDonochVA Fixed. I moved to the `feature-recipe-edit-dialog.component.ts` itself. It listens to `RecipesApiActions.createRecipeSuccess` and `RecipesApiActions.editRecipeSuccess` actions.